pkg/client/kubernetes/types.go

```diff
@@ -245,6 +246,10 @@ type Interface interface {
 
 	// Deployments
 	// Deprecated: Use `Client()` and utils instead.
+	CreateDeploymentObject(*appsv1.Deployment, bool) (*appsv1.Deployment, error)
```

Comment:
Can't we directly make use of our controller-runtime `Client` here instead of exposing yet other deprecated methods?